src/deepsparse/loggers/prometheus_logger.py

```diff
@@ -136,14 +136,51 @@ def _setup_client(self):
     def _validate(self, value: Any) -> Any:
         # make sure we are passing a value that is
         # a valid metric by prometheus client's standards
-        if not isinstance(value, SUPPORTED_DATA_TYPES):
+        if not isinstance(value, _SUPPORTED_DATA_TYPES):
             raise ValueError(
                 "Prometheus logger expects the incoming values "
-                f"to be one of the type: {SUPPORTED_DATA_TYPES}"
+                f"to be one of the type: {_SUPPORTED_DATA_TYPES}"
             )
         return value
 
 
+def get_prometheus_metric(
+    identifier: str,
+    category: MetricCategories,
+    registry: CollectorRegistry,
+    description_template: str = _DESCRIPTION,
+) -> Optional["MetricWrapperBase"]:  # noqa: F821
+    """
+    Get a Prometheus metric object for the given identifier and category.
+
+    :param identifier: The name of the thing that is being logged.
+    :param category: The metric category that the log belongs to
+    :param registry: The Prometheus registry to which the metric should be added
+    :param description_template: The template for the description of the metric
+    :return: The Prometheus metric object or None if the identifier not supported
+    """
+    metric = Summary
+
+    return metric(
+        format_identifier(identifier),
+        description_template.format(
```

Comment:
do we expect this value to be overwritten? if so, then it would need to be overwritten with the exact same template variables or this will break

Reply:
No, this value should not be overwritten and the logic protects it from getting overwritten.

In general, whenever the `PrometheusLogger` receives a log (identifier) it will:
a) Either not find it in the "registry" (self._prometheus_metrics) that maps identifiers to Prometheus metrics and go through the appropriate composition of functions to create a metric and save it to the "registry". The composition of functions in that case would be: `self.log` -> `self._get_prometheus_metric` -> `self._add_metric_to_registry` -> `get_prometheus_metric`
b) Find it in the "registry", if it has been previously seen. Then the logic only goes through `self.log` -> `self._get_prometheus_metric`

Reply:
if it should not be overwritten, then why do we have a variable for it in the function rather than reference the constant `_DESCRIPTION` directly? also let's rename to `_METRIC_DESCRIPTION_TEMPLATE`

Reply:
Good point